ocaml/testsuite/tests/typing-layouts/erasable_annot.ml

```diff
@@ -0,0 +1,241 @@
+(* TEST
```

Comment:
Nice tests here. Please also add for
* `... int as (_ : immediate) ...` (should be accepted, because that annotation does nothing)
* `type 'a id = 'a` and then `... ('a id as (_ : immediate)) ...` (should be rejected, as it will affect `'a`)

Reply:
Added both test cases